src/app/core/models/templates/team-accepted-template.model.ts

```diff
@@ -0,0 +1,22 @@
+import { GithubComment } from '../github/github-comment.model';
+import { Header, Template } from './template.model';
+
+const TeamAcceptedRegexString = 'response not required for this bug as the team has accepted the bug as it';
+export const TeamAcceptedMessage = 'Your ' + TeamAcceptedRegexString + ' is.';
+export const TeamAcceptedHeader = { teamAccepted: new Header(TeamAcceptedRegexString, 0) };
```

Comment:
This is a bit funky because of how `Header` and the regex in `Template` works.
Constructing a Header will automatically prepend a space in front of the string (ordinarily this is to separate the # symbols from the actual header title), but if there's no # symbols then that creates a problem with the regex matching.
In addition, the regex expression constructed in `Template` is in the form 
`(${headerString})\\s+([\\s\\S]*?)(?=${headerString}|$)`
where it expects at least one whitespace character after the header string (that's the `\\s+` part).

If we use the original team accepted message
`Your response not required for this bug as the team has accepted the bug as it is.`
The regex won't have any matches because there isn't a space in front of the `Your` and there's no space after the `is.` in the actual comment. Which is why for the regex used, the first and last words are removed to work around that.

Reply:
Don't go too far with regular expressions as they cause too much headaches to debug. Furthermore, regex can be slow and not all future dev can be expected to know regex well. For example, a `if x in y and z in y` is a lot easier to understand than a complicated regular expression.

Reply:
@damithc As discussed earlier, regex is built into how `Template` works, so moving away from regex would mean changing quite a bit of the code. 

> Another way we can go about it is to fix the code within `template.model.ts`.
> Such that when our `headerHash` is empty, we should not append a whitespace in front of it.

This should solve the issue of leaving out the "Your " in the phrase, and is pretty simple to do - just change the `toString()` method of the `Header` class

If we want to include the " is." in the regex phrase, then one really hacky way to do it is to add a space add the back of `comment.body`, i.e.

```typescript
const teamAcceptedComment = githubIssueComments.find((comment) => this.test(comment.body + ' '));
```

so that the part of the regex that checks for whitespace at the end of the header string will be satisfied. Unfortunately `Header` and `Template` wasn't designed for this situation, so we either do this or we omit the last word of the `TeamAcceptedMessage` (unless there's a third way that I haven't thought of)

Reply:
> 
> Such that when our `headerHash` is empty, we should not append a whitespace in front of it.
> 
> Any thoughts on this? @damithc @dingyuchen @kkangs0226

I think the underlying issue is that the regex is attempting to match on multiple `Header`s, with the assumption of the format 

```
<header1>
<content1>
<header2>
<content2>
.
```
If we allow just the `headerString` to match followed by the EOS terminal, this should cover the case. (together with not prepending a space) ie.
`(${headerString})(\\s+|$)([\\s\\S]*?)(?=${headerString}|$)`
The acceptance header would also be less hacky.

But ultimately we should investigate the places this regex is being used, and do some refactoring, in another branch, since we are nearing the code freeze.